ipynb/filterEMP.ipynb

```diff
@@ -0,0 +1,254 @@
+    "    counts_subset = counts.filter(toBeFiltered_idx, axis='sample', inplace=False)\n",
+    "    \n",
+    "    #create the new filename\n",
+    "    print(\"Writing resulting biom table to '\"+ targetFile +\"'\")\n",
+    "    with biom_open(targetFile, 'w') as f:\n",
+    "        counts_subset.to_hdf5(f, \"EMP sub-set generation notebook\")\n",
+    "        \n",
+    "    print(\"Generating summary.\")\n",
+    "    inp = targetFile\n",
+    "    out = targetFile + \".summary.txt\"\n",
+    "    !biom summarize-table -i \"$inp\" > \"$out\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Reads a (beta diversity) distance matrix from 'sourceFile' and returns only those cells that belong to samples that are listed in 'sampleIDsToKeep'. Output is written to 'targetFile'\n",
+    "\n",
+    "def filterBetadivMatrix(sampleIDsToKeep, sourceFile, targetFile):\n",
+    "    out = open(targetFile, \"w\")\n",
+    "    f = open(sourceFile, \"r\")\n",
+    "    columnsToKeep = []\n",
+    "    foundIdx = []\n",
+    "    lidx = list(sampleIDsToKeep)\n",
+    "    matrixIdx = f.readline().rstrip().split(\"\\t\")\n",
+    "    for i, sid in enumerate(matrixIdx):\n",
+    "        if sid in lidx:\n",
+    "            columnsToKeep.append(i)\n",
+    "            foundIdx.append(sid)\n",
+    "            out.write(\"\\t\" + sid)\n",
+    "    out.write(\"\\n\")\n",
+    "    \n",
+    "    notFound = list(set(sampleIDsToKeep) - set(foundIdx))\n",
+    "    print(\"Sub-sampling down to \" + str(len(sampleIDsToKeep)) + \" of which \" + str(len(notFound)) + \" are not in the input matrix.\")\n",
+    "    \n",
+    "    for lineCount, line in enumerate(f):\n",
+    "        fields = line.rstrip().split(\"\\t\")\n",
+    "        if fields[0] in lidx:\n",
+    "            out.write(fields[0])\n",
+    "            for i in columnsToKeep:\n",
+    "                out.write(\"\\t\" + fields[i])\n",
+    "            out.write(\"\\n\")\n",
+    "        if (lineCount % int(len(matrixIdx) / 10) == 0):\n",
+    "            print(\"  processed \" + str(int(lineCount / len(matrixIdx) * 100)) + \" %\")\n",
+    "            \n",
+    "    print(\"wrote resulting distance matrix to '\" +targetFile+ \"'\")\n",
+    "    out.close()\n",
+    "    f.close()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "#we load the mapping file into a Pandas dataframe\n",
+    "metadata = pd.read_csv(file_mapping, sep=\"\\t\", index_col=0, low_memory=False)\n",
+    "\n",
+    "#we chose a (sub-)set of sample IDs which we want to have in our output set. E.g. all sampleIDs that have a np.True_ in the 'subset_2000' column, i.e. those 2000 samples that form the EMP 2000 sub-set \n",
+    "idx = metadata[metadata.subset_2000 == np.True_].index\n",
```

Comment:
maybe for clarity change `metadata.subset_200` to `metadata['subset_200']`? It's easier to spot